pkg/controller/installation/products/threescale/reconciler.go

```diff
@@ -305,39 +311,165 @@ func (r *Reconciler) reconcileComponents(ctx context.Context, serverClient pkgcl
+		return v1alpha1.PhaseAwaitingComponents, nil
+	}
 
-	logrus.Info("Creating blob storage")
-	_, err := controllerutil.CreateOrUpdate(ctx, serverClient, blobStorage, func(existing runtime.Object) error {
-		c := existing.(*crov1.BlobStorage)
-		c.Spec.Type = r.installation.Spec.Type
-		c.Spec.Tier = "production"
-		c.Spec.SecretRef = &crov1.SecretRef{
-			Name:      fmt.Sprintf("3scale-s3-%s", r.installation.Name),
-			Namespace: r.installation.Namespace,
-		}
+	// get the secret created by the cloud resources operator
+	// containing system redis connection details
+	systemCredSec := &v1.Secret{}
+	err = serverClient.Get(ctx, pkgclient.ObjectKey{Name: systemRedis.Status.SecretRef.Name, Namespace: systemRedis.Status.SecretRef.Namespace}, systemCredSec)
+	if err != nil {
+		return v1alpha1.PhaseFailed, errors.Wrap(err, "failed to get system redis credential secret")
+	}
+
+	// create system redis external connection secret needed for the 3scale apimanager
+	redisSecret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      externalRedisSecretName,
+			Namespace: r.Config.GetNamespace(),
+		},
+		Data: map[string][]byte{},
+	}
+	_, err = controllerutil.CreateOrUpdate(ctx, serverClient, redisSecret, func(existing runtime.Object) error {
+		uri := systemCredSec.Data["uri"]
+		port := systemCredSec.Data["port"]
+		conn := fmt.Sprintf("redis://%s:%s/1", uri, port)
+		redisSecret.Data["URL"] = []byte(conn)
+		redisSecret.Data["MESSAGE_BUS_URL"] = []byte(conn)
 		return nil
 	})
 	if err != nil {
-		return v1alpha1.PhaseFailed, errors.Wrap(err, "failed to reconcile blob storage request")
+		return v1alpha1.PhaseFailed, errors.Wrapf(err, "failed to create or update 3scale %s connection secret", externalRedisSecretName)
 	}
 
-	// wait for the blob storage cr to reconcile
-	if blobStorage.Status.Phase != crov1.PhaseComplete {
+	// wait for the postgres cr to reconcile
+	if postgres.Status.Phase != crov1.PhaseComplete {
 		return v1alpha1.PhaseAwaitingComponents, nil
 	}
 
+	// get the secret containing redis credentials
+	postgresCredSec := &v1.Secret{}
+	err = serverClient.Get(ctx, pkgclient.ObjectKey{Name: postgres.Status.SecretRef.Name, Namespace: postgres.Status.SecretRef.Namespace}, postgresCredSec)
+	if err != nil {
+		return v1alpha1.PhaseFailed, errors.Wrap(err, "failed to get postgres credential secret")
+	}
+
+	// create postgres external connection secret
+	postgresSecret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      externalPostgresSecretName,
+			Namespace: r.Config.GetNamespace(),
+		},
+		Data: map[string][]byte{},
+	}
+	_, err = controllerutil.CreateOrUpdate(ctx, serverClient, postgresSecret, func(existing runtime.Object) error {
+		username := postgresCredSec.Data["username"]
+		password := postgresCredSec.Data["password"]
+		url := fmt.Sprintf("postgresql://%s:%s@%s:5432/%s", username, password, postgresCredSec.Data["host"], postgresCredSec.Data["database"])
```

Comment:
/LGTM 
May want to apply a var to `5432`, I know is the default postgres port

Reply:
@dimitraz do we not expose the port as part of the secret? if not, we should create a ticket for that

Reply:
We do, I just missed that. Updated now :+1: